Formula/ada-url.rb

```diff
@@ -0,0 +1,36 @@
+class AdaUrl < Formula
+  desc "WHATWG-compliant and fast URL parser written in modern C++"
+  homepage "https://github.com/ada-url/ada"
+  url "https://github.com/ada-url/ada/archive/refs/tags/v2.2.0.tar.gz"
+  sha256 "40a6b3fe0d5c62936c081e8403790ec05d5afe3d0909eece894efcfef7e678ee"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/ada-url/ada.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.11" => :build
+  depends_on macos: :catalina
+
+  def install
+    system "cmake", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+      #include "ada.h"
+      #include <iostream>
+
+      int main(int , char *[]) {
+        auto url = ada::parse<ada::url_aggregator>("https://www.github.com/ada-url/ada");
+        url->set_protocol("http");
+        std::cout << url->get_protocol() << std::endl;
+        return EXIT_SUCCESS;
+      }
+    EOS
+
+    system ENV.cxx, "test.cpp", "-std=c++17",
+           "-I#{include}", "-L#{lib}", "-lada", "-o", "test"
+    assert_equal "\"Homebrew\"\n", shell_output("./test")
```

Comment:
Seems like this is what you should be expecting?
```suggestion
    assert_equal "http", shell_output("./test").chomp
```